File: Codeforces/344A_Magnets.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    int n;
    vector <string> magnets;

    cin >> n;

    while (n > 0) {
        string magnet;
        cin >> magnet;
        magnets.push_back(magnet);
        n--;
    }

    int groups = 1;

    for (int i = 0; i < magnets.size(); i++) {
        if (i != magnets.size() - 1) {
            if (magnets[i][1] == magnets[i+1][0])
                groups++;
        }
    }

    cout << groups << endl;

    return 0;
}

File: Codeforces/344A_Magnets.py
n = int(input())
magnets = []
groups = 1

while n > 0:
    magnets.append(input())
    n -= 1

for i in range(len(magnets)):
    if i != len(magnets) - 1:
        if magnets[i][1] == magnets[i+1][0]:
            groups += 1

print(groups)
